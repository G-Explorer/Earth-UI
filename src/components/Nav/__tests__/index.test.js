import React from 'react'
import { render } from 'enzyme'
import { Nav, SubNav, NavItemGroup, NavItem } from '..'

describe('<Nav>', () => {
  const navClassName = 'true-nav'
  it('should render successfully when there are only NavItems in Nav', () => {
    const wrapper = (
      <Nav selectedId="1">
        <NavItem id="1" />
        <NavItem id="2" />
      </Nav>
    )
    expect(render(wrapper)).toMatchSnapshot()
  })
  it('should render successfully when there are SubNavs in Nav', () => {
    const wrapper = (
      <Nav selectedId="1">
        <SubNav>
          <NavItem id="2" />
          <NavItem id="3" />
        </SubNav>
      </Nav>
    )
    expect(render(wrapper)).toMatchSnapshot()
  })
  it('should render successfully when there are NavItemGroup in SubNav', () => {
    const wrapper = (
      <Nav selectedId="1">
        <NavItem id="1" />
        <NavItem id="2" />
        <SubNav>
          <NavItem id="3" />
          <NavItem id="4" />
        </SubNav>
        <SubNav>
          <NavItemGroup>
            <NavItem id="4" />
            <NavItem id="5" />
          </NavItemGroup>
        </SubNav>
      </Nav>
    )
    expect(render(wrapper)).toMatchSnapshot()
  })
  it('should keep Nav collapsed when there is no collapsed prop on Nav', () => {
    const wrapper = (
      <Nav selectedId="1">
        <NavItem id="1" />
      </Nav>
    )
    expect(render(wrapper).find(`.${navClassName}--collapsed`)).toBeTruthy()
  })
  it('should keep Nav open when set collapsed prop to be false on Nav', () => {
    const wrapper = (
      <Nav selectedId="1" collapsed>
        <NavItem id="1" />
      </Nav>
    )
    expect(
      render(wrapper)
        .find(`.${navClassName}`)
        .hasClass(`.${navClassName}--collapsed`)
    ).toBeFalsy()
  })
})
