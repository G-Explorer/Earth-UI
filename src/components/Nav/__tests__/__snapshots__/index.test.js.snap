// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Nav> should render successfully when there are NavItemGroup in SubNav 1`] = `
<div
  class="true-nav true-nav--collapsed"
>
  <div
    class="true-nav__top"
  >
    <div
      class="true-nav__top-left-container"
    >
      <svg
        class="true-nav__top-left-container-menu-icon"
        height="22"
        width="22"
      >
        <g
          fill="#FFF"
          fill-rule="evenodd"
        >
          <path
            d="M0 1.01A1 1 0 0 1 1.002 0h19.996A1 1 0 0 1 22 1.01v1.98A1 1 0 0 1 20.998 4H1.002A1 1 0 0 1 0 2.99V1.01z"
          />
          <rect
            height="4"
            rx="1"
            width="22"
            y="9"
          />
          <rect
            height="4"
            rx="1"
            width="22"
            y="18"
          />
        </g>
      </svg>
    </div>
    <div
      class="true-nav__top-center-container"
    >
      <img
        alt="Logo"
        class="true-nav__top-center-container-logo"
      />
    </div>
    <div
      class="true-nav__top-right-container"
    >
      <button
        class="true-button true-button--md true-button__icon-only true-nav__btn-icon true-nav__top-right-container-close-icon"
        type="button"
      >
        <i
          class="true-icon true-icon__close"
        />
      </button>
    </div>
  </div>
  <div
    class="true-scrollbar true-nav__scrollbar"
  >
    <div
      style="position:relative;overflow:hidden;width:100%;height:100%"
    >
      <div
        style="position:absolute;top:0;left:0;right:0;bottom:0;overflow:scroll;-webkit-overflow-scrolling:touch;margin-right:0;margin-bottom:0"
      >
        <ul>
          <li
            class="true-nav__item true-nav__item--active"
            id="1"
          >
            <div
              class="true-nav_item-title"
            >
              <i
                class="true-icon true-nav__item-icon"
              />
              <span />
            </div>
          </li>
          <li
            class="true-nav__item"
            id="2"
          >
            <div
              class="true-nav_item-title"
            >
              <i
                class="true-icon true-nav__item-icon"
              />
              <span />
            </div>
          </li>
          <li
            class="true-nav__sub-nav"
          >
            <div
              class="true-nav_sub-nav-title"
            >
              <i
                class="true-icon true-nav__sub-nav-title-icon"
              />
              <span>
                <i
                  class="true-icon true-icon__triangleright true-nav__sub-nav-title-toggle"
                />
              </span>
            </div>
          </li>
          <li
            class="true-nav__sub-nav"
          >
            <div
              class="true-nav_sub-nav-title"
            >
              <i
                class="true-icon true-nav__sub-nav-title-icon"
              />
              <span>
                <i
                  class="true-icon true-icon__triangleright true-nav__sub-nav-title-toggle"
                />
              </span>
            </div>
          </li>
        </ul>
      </div>
      <div
        style="position:absolute;height:6px;transition:opacity 200ms;opacity:0;display:none;right:2px;bottom:2px;left:2px;border-radius:3px"
      >
        <div
          style="position:relative;display:block;height:100%;cursor:pointer;border-radius:inherit;background-color:rgba(0,0,0,.2)"
        />
      </div>
      <div
        style="position:absolute;width:6px;transition:opacity 200ms;opacity:0;display:none;right:2px;bottom:2px;top:2px;border-radius:3px"
      >
        <div
          class="thumb-vertical"
          style="position:relative;display:block;width:100%;background-color:rgba(90, 90, 90, .8);border-radius:inherit"
        />
      </div>
    </div>
  </div>
  <div
    class="true-nav__bottom"
  >
    <div
      class="true-nav__bottom-image"
    >
      <img
        alt="Avatar"
        class="true-nav__bottom-image-icon"
      />
    </div>
    <div
      class="true-nav__bottom-user"
    >
      <span
        class="true-nav__bottom-user-name"
      />
      <span
        class="true-nav__bottom-user-company"
      />
    </div>
    <div
      class="true-nav__bottom-logout"
    >
      <button
        class="true-button true-button--md true-button__icon-only true-nav__btn-icon true-nav__bottom-logout-icon"
        type="button"
      >
        <i
          class="true-icon true-icon__logout"
        />
      </button>
    </div>
    <div
      class="true-nav__bottom-settings"
    >
      <button
        class="true-button true-button--md true-button__icon-only true-nav__btn-icon true-nav__bottom-settings-icon"
        type="button"
      >
        <i
          class="true-icon true-icon__settings"
        />
      </button>
    </div>
  </div>
</div>
`;

exports[`<Nav> should render successfully when there are SubNavs in Nav 1`] = `
<div
  class="true-nav true-nav--collapsed"
>
  <div
    class="true-nav__top"
  >
    <div
      class="true-nav__top-left-container"
    >
      <svg
        class="true-nav__top-left-container-menu-icon"
        height="22"
        width="22"
      >
        <g
          fill="#FFF"
          fill-rule="evenodd"
        >
          <path
            d="M0 1.01A1 1 0 0 1 1.002 0h19.996A1 1 0 0 1 22 1.01v1.98A1 1 0 0 1 20.998 4H1.002A1 1 0 0 1 0 2.99V1.01z"
          />
          <rect
            height="4"
            rx="1"
            width="22"
            y="9"
          />
          <rect
            height="4"
            rx="1"
            width="22"
            y="18"
          />
        </g>
      </svg>
    </div>
    <div
      class="true-nav__top-center-container"
    >
      <img
        alt="Logo"
        class="true-nav__top-center-container-logo"
      />
    </div>
    <div
      class="true-nav__top-right-container"
    >
      <button
        class="true-button true-button--md true-button__icon-only true-nav__btn-icon true-nav__top-right-container-close-icon"
        type="button"
      >
        <i
          class="true-icon true-icon__close"
        />
      </button>
    </div>
  </div>
  <div
    class="true-scrollbar true-nav__scrollbar"
  >
    <div
      style="position:relative;overflow:hidden;width:100%;height:100%"
    >
      <div
        style="position:absolute;top:0;left:0;right:0;bottom:0;overflow:scroll;-webkit-overflow-scrolling:touch;margin-right:0;margin-bottom:0"
      >
        <ul>
          <li
            class="true-nav__sub-nav"
          >
            <div
              class="true-nav_sub-nav-title"
            >
              <i
                class="true-icon true-nav__sub-nav-title-icon"
              />
              <span>
                <i
                  class="true-icon true-icon__triangleright true-nav__sub-nav-title-toggle"
                />
              </span>
            </div>
          </li>
        </ul>
      </div>
      <div
        style="position:absolute;height:6px;transition:opacity 200ms;opacity:0;display:none;right:2px;bottom:2px;left:2px;border-radius:3px"
      >
        <div
          style="position:relative;display:block;height:100%;cursor:pointer;border-radius:inherit;background-color:rgba(0,0,0,.2)"
        />
      </div>
      <div
        style="position:absolute;width:6px;transition:opacity 200ms;opacity:0;display:none;right:2px;bottom:2px;top:2px;border-radius:3px"
      >
        <div
          class="thumb-vertical"
          style="position:relative;display:block;width:100%;background-color:rgba(90, 90, 90, .8);border-radius:inherit"
        />
      </div>
    </div>
  </div>
  <div
    class="true-nav__bottom"
  >
    <div
      class="true-nav__bottom-image"
    >
      <img
        alt="Avatar"
        class="true-nav__bottom-image-icon"
      />
    </div>
    <div
      class="true-nav__bottom-user"
    >
      <span
        class="true-nav__bottom-user-name"
      />
      <span
        class="true-nav__bottom-user-company"
      />
    </div>
    <div
      class="true-nav__bottom-logout"
    >
      <button
        class="true-button true-button--md true-button__icon-only true-nav__btn-icon true-nav__bottom-logout-icon"
        type="button"
      >
        <i
          class="true-icon true-icon__logout"
        />
      </button>
    </div>
    <div
      class="true-nav__bottom-settings"
    >
      <button
        class="true-button true-button--md true-button__icon-only true-nav__btn-icon true-nav__bottom-settings-icon"
        type="button"
      >
        <i
          class="true-icon true-icon__settings"
        />
      </button>
    </div>
  </div>
</div>
`;

exports[`<Nav> should render successfully when there are only NavItems in Nav 1`] = `
<div
  class="true-nav true-nav--collapsed"
>
  <div
    class="true-nav__top"
  >
    <div
      class="true-nav__top-left-container"
    >
      <svg
        class="true-nav__top-left-container-menu-icon"
        height="22"
        width="22"
      >
        <g
          fill="#FFF"
          fill-rule="evenodd"
        >
          <path
            d="M0 1.01A1 1 0 0 1 1.002 0h19.996A1 1 0 0 1 22 1.01v1.98A1 1 0 0 1 20.998 4H1.002A1 1 0 0 1 0 2.99V1.01z"
          />
          <rect
            height="4"
            rx="1"
            width="22"
            y="9"
          />
          <rect
            height="4"
            rx="1"
            width="22"
            y="18"
          />
        </g>
      </svg>
    </div>
    <div
      class="true-nav__top-center-container"
    >
      <img
        alt="Logo"
        class="true-nav__top-center-container-logo"
      />
    </div>
    <div
      class="true-nav__top-right-container"
    >
      <button
        class="true-button true-button--md true-button__icon-only true-nav__btn-icon true-nav__top-right-container-close-icon"
        type="button"
      >
        <i
          class="true-icon true-icon__close"
        />
      </button>
    </div>
  </div>
  <div
    class="true-scrollbar true-nav__scrollbar"
  >
    <div
      style="position:relative;overflow:hidden;width:100%;height:100%"
    >
      <div
        style="position:absolute;top:0;left:0;right:0;bottom:0;overflow:scroll;-webkit-overflow-scrolling:touch;margin-right:0;margin-bottom:0"
      >
        <ul>
          <li
            class="true-nav__item true-nav__item--active"
            id="1"
          >
            <div
              class="true-nav_item-title"
            >
              <i
                class="true-icon true-nav__item-icon"
              />
              <span />
            </div>
          </li>
          <li
            class="true-nav__item"
            id="2"
          >
            <div
              class="true-nav_item-title"
            >
              <i
                class="true-icon true-nav__item-icon"
              />
              <span />
            </div>
          </li>
        </ul>
      </div>
      <div
        style="position:absolute;height:6px;transition:opacity 200ms;opacity:0;display:none;right:2px;bottom:2px;left:2px;border-radius:3px"
      >
        <div
          style="position:relative;display:block;height:100%;cursor:pointer;border-radius:inherit;background-color:rgba(0,0,0,.2)"
        />
      </div>
      <div
        style="position:absolute;width:6px;transition:opacity 200ms;opacity:0;display:none;right:2px;bottom:2px;top:2px;border-radius:3px"
      >
        <div
          class="thumb-vertical"
          style="position:relative;display:block;width:100%;background-color:rgba(90, 90, 90, .8);border-radius:inherit"
        />
      </div>
    </div>
  </div>
  <div
    class="true-nav__bottom"
  >
    <div
      class="true-nav__bottom-image"
    >
      <img
        alt="Avatar"
        class="true-nav__bottom-image-icon"
      />
    </div>
    <div
      class="true-nav__bottom-user"
    >
      <span
        class="true-nav__bottom-user-name"
      />
      <span
        class="true-nav__bottom-user-company"
      />
    </div>
    <div
      class="true-nav__bottom-logout"
    >
      <button
        class="true-button true-button--md true-button__icon-only true-nav__btn-icon true-nav__bottom-logout-icon"
        type="button"
      >
        <i
          class="true-icon true-icon__logout"
        />
      </button>
    </div>
    <div
      class="true-nav__bottom-settings"
    >
      <button
        class="true-button true-button--md true-button__icon-only true-nav__btn-icon true-nav__bottom-settings-icon"
        type="button"
      >
        <i
          class="true-icon true-icon__settings"
        />
      </button>
    </div>
  </div>
</div>
`;

exports[`Nav should render correctly 1`] = `
<div
  class="true-nav"
>
  <ul>
    <li
      class="true-nav__sub-nav true-nav__sub-nav_open"
      id="2"
    >
      <div
        class="true-nav_sub-nav-entity"
        style="padding-left:24px"
      >
        <i
          class="true-icon true-icon__triangleright true-nav__sub-nav-toggle"
        />
      </div>
      <ul>
        <li
          class="true-nav__item-group"
        >
          <div
            class="true-nav__item-group-title"
            style="padding-left:32px"
          />
          <ul>
            <li
              class="true-nav__item"
              id="3"
            >
              <div
                class="true-nav_item-entity"
                style="padding-left:48px"
              />
            </li>
            <li
              class="true-nav__item"
              id="4"
            >
              <div
                class="true-nav_item-entity"
                style="padding-left:48px"
              />
            </li>
          </ul>
        </li>
      </ul>
    </li>
  </ul>
</div>
`;
